### PR TITLE
stm32h5: introduce the new stm32H5xx serie

### PR DESCRIPTION
**Pull request by FRASTM**
This is the  first delivery to zephyr hal_stm32
    from the https://github.com/STMicroelectronics/STM32CubeH5/
The STM32H503xx devices are a high-performance microcontrollers family (STM32H5 Series) based on the high-performance Arm® Cortex®-M33 32-bit RISC core. They operate at a frequency of up to 250 MHz.

--> new  stm32Cube/stm32H5 and Pin description for the stm32H5xx devices.

### CODE REVIEW

File: stm32cube/stm32h5xx/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA_EX drivers/src/stm32h5xx_hal_dma_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DSI drivers/src/stm32h5xx_hal_dsi.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DTS drivers/src/stm32h5xx_hal_dts.c)
+if(CONFIG_ETH_STM32_HAL_API_V2)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/stm32h5xx_hal_eth.c)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/stm32h5xx_hal_eth_ex.c)
+else()
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/Legacy/stm32h5xx_hal_eth.c)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/Legacy/stm32h5xx_hal_eth_ex.c)
+endif()
```
Comment (erwango) [inline stm32cube/stm32h5xx/CMakeLists.txt:36]: These files don't exist.
API V1 is not available on H5 series

**Review by erwango — CHANGES_REQUESTED** [review]
Missing files in stm32cube/common_ll/include

File: stm32cube/stm32h5xx/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DAC_EX drivers/src/stm32h5xx_hal_dac_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DCACHE drivers/src/stm32h5xx_hal_dcache.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DCMI drivers/src/stm32h5xx_hal_dcmi.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA drivers/src/stm32h5xx_hal_dma.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA_EX drivers/src/stm32h5xx_hal_dma_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DTS drivers/src/stm32h5xx_hal_dts.c)
+if(CONFIG_ETH_STM32_HAL_API_V2)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/stm32h5xx_hal_eth.c)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/stm32h5xx_hal_eth_ex.c)
+endif()
```
Comment (erwango) [inline stm32cube/stm32h5xx/CMakeLists.txt:29]: If there's a dependency, better make it explicit, than silently fail if not satisfied

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: stm32cube/stm32h5xx/CMakeLists.txt (hunk 2 above)
Comment (erwango) [reply stm32cube/stm32h5xx/CMakeLists.txt:29]: Can you try something like :
if (CONFIG_USE_STM32_HAL_ETH)
    if (NOT CONFIG_ETH_STM32_HAL_API_V2)
         message(FATAL_ERROR "command failed with return code: ${ret}")
   endif()
else
    if(CONFIG_ETH_STM32_HAL_API_V2)
	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/stm32h5xx_hal_eth.c)
	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/stm32h5xx_hal_eth_ex.c)
    endif()
endif()

File: stm32cube/stm32h5xx/drivers/include/stm32_assert.h
Comment (str4t0m) [inline stm32cube/stm32h5xx/drivers/include/stm32_assert.h:1]: Shouldn't this file just include `#include <stm32h5xx_hal_conf.h>` to be in line with the other series?

File: stm32cube/stm32h5xx/CMakeLists.txt (hunk 2 above)
Comment (FRASTM) [reply stm32cube/stm32h5xx/CMakeLists.txt]: adding message for the a wrong config: CONFIG_USE_STM32_HAL_ETH but NOT CONFIG_ETH_STM32_HAL_API_V2

**Review by erwango — APPROVED** [review]

File: stm32cube/stm32h5xx/README [hunk 3]
```diff
@@ -48,4 +54,10 @@ Patch List:
       . renamed stm32h5xx_hal_timebase_rtc_wakeup_template. c to stm32h5xx_hal_timebase_rtc_wakeup.c
       . renamed stm32h5xx_hal_timebase_tim_template. c to stm32h5xx_hal_timebase_tim.c
 
+   *Var init in LL_I2S_Init
+    - In the LL_I2S_Init function, initialisation of the sourceclock to avoid compilation warning.
+      Impacted files:
+       drivers/include/stm32h5xx_ll_spi.h
+      ST internal bug : 147754
+
```
Comment (gautierg-st) [inline stm32cube/stm32h5xx/README:62]: This patch is described twice, Please remove one of them

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**